### PR TITLE
Death is Our Destination (New funeral rite)

### PR DESCRIPTION
**Pull request by Mycah142**
## About The Pull Request
Since I was on this weird death kick, someone mentioned Iriska never fit into a coffin. Well, that's a crying shame! So I made this wide boy
84dv5y
And while I was at THAT, I decided to make coffins burnable, and have that be a funeral rite. It's more effective than the morgue, requiring a bit more effort than drag body to tray, click. I made coffins a bit less closet-like as well, requiring tools to open or close, requiring a bit of time to do so; they also only accept mobs now, so make sure Jimmy has his brain in before you torch 'im. Y'know, for flavor.

## Why It's Good For The Game
Captain COOOOOOOOOOOOPIUM can now hold a midnight vigil for the cat his FO purposefully shot trying to steal the Destiny.

## Testing
Changes to the coffin were pretty high, and testing it local was difficult, so I used a video instead.
[dreamseeker_KcdFSlAx8L.webm](https://github.com/discordia-space/CEV-Eris/assets/11076040/8ca1c360-de74-4ddc-ba04-e2d7621e4dc9)
(I don't know why it just refuses to embed)

## Changelog
:cl:
add: Pauper's Coffin, a larger coffin.
add: Funeral Pyre capabilities, burning coffins.
tweak: Coffins now require tools to open/close. Can still be walked out of.
tweak: Coffins now only accept mobs.
imageadd: added the worst flame.gif I could summon.
/:cl:

### CODE REVIEW

**Comment by thevandie** [comment]
can you also fix the thing where the funeral things ONLY apply when your player ghost is in the body/brain?

**Comment by Mycah142** [comment]
> can you also fix the thing where the funeral things ONLY apply when your player ghost is in the body/brain?

Watch the video. I'm clearly not in the body/brain.

**Comment by thevandie** [comment]
didn't know there was a video, your recording system is archaic and didn't embed
mah bad

**Comment by SirRichardFrancis** [comment]
> didn't embed mah

[279232849-e177e6c7-5c72-4c47-be64-29e870453ffa.webm](https://github.com/discordia-space/CEV-Eris/assets/65828539/365ebf7c-4743-41d6-a846-e7bfdf84320c)

**Comment by Mycah142** [comment]
Currently, the Morgue looks for a mob. If the Mob lacks a brain, then it no longer has an associated Ckey, as removing the brain moves the Ckey to the brain.

So if medical personnel morgue the body, but have the brain in the bag with them, it also doesn't work, as it specifically looks for a mob, and brains are not mobs.

**Comment by SirRichardFrancis** [comment]
Fire animation is really janky. Consider at least making it slower, that should help somewhat.

**Comment by Mycah142** [comment]
> Fire animation is really janky. 
wait til you see my code

File: code/game/objects/structures/crates_lockers/closets/coffin.dm [hunk 1]
```diff
@@ -9,6 +9,14 @@
 	spawn_tags = SPAWN_TAG_CLOSET_COFFIN
 	bad_type = /obj/structure/closet/coffin
 	var/mob/living/occupant = null
+	var/on_fire = 0
```
Comment (SirRichardFrancis) [inline code/game/objects/structures/crates_lockers/closets/coffin.dm:12]:
```suggestion
	var/on_fire = FALSE
```

**Review by SirRichardFrancis — COMMENTED** [review]
> wait til you see my code

comehere

**Comment by thevandie** [comment]
btw, the pauper coffin doesn't have an open door sprite from what I can see, probably should

image

even if it's the incorrect size, there should still be something there

**Comment by Mycah142** [comment]
I am aware it lacks a door. There's two pixels of height where the door should be.  The coffin stays chonk. Door be damned.

**Comment by thevandie** [comment]
image

would this not work?

**Comment by SirRichardFrancis** [comment]
You can add a door as an overlay, just offset it lol.

**Comment by Mycah142** [comment]
It's settled I'll make the door a new object you gotta pull off.

**Comment by SirRichardFrancis** [comment]
No no no, don't have to be a separate object. Just slap a second sprite on top.

**Comment by thevandie** [comment]
image

but yeah, this is literally all you'd need, just a shaved version of the normal one, to make it look like it's slumped over the back.
like this random stock image

image

**Review by Firefox13 — CHANGES_REQUESTED** [review]
Atomize your PRs. Fixes in one, new features in the other.

**Review by Firefox13 — COMMENTED** [review]
Can you resist out of a closed coffin? Is there any way to acquire puper's coffin? Are you sure people aren't going to misuse the its humongous storage capacity?

**Comment by Mycah142** [comment]
> Can you resist out of a closed coffin?

Currently, without tools, no. Is there a way to resist out of a welded closet?

>Is there any way to acquire puper's coffin?

You right I forgot to make the crafting recipe.

>Are you sure people aren't going to misuse the its humongous storage capacity?

I can make it bodies only easy.

**Comment by Mycah142** [comment]
> > Can you resist out of a closed coffin?
> 
> Currently, without tools, no. Is there a way to resist out of a welded closet?
> 

I could make it only work if the occupant is dead.

**Comment by Firefox13** [comment]
>Currently, without tools, no. Is there a way to resist out of a welded closet?

изображение

Even if this code doesn't work you shouldn't be able to permalock people like this.

**Comment by thevandie** [comment]
I do believe you can break out of welded closets, yeah
Takes 5 minutes and it's reset when the locker is moved, but it does work

Btw, like I said I can't access the discord currently, but if you're cuffed and you rest, you're physically unable to get up. It cancels the loading bar every time.

**Comment by Mycah142** [comment]
> Can you resist out of a closed coffin?

Hey fun fact. One can just fucking walk and the coffin opens, like a closet. I completely fucking spaced it on giving it a welded status, so anyone trapped inside can easily escape it if they're conscious. I assume this is acceptable.

**Comment by Mycah142** [comment]
I did not mean to fukken close this PR I fatfingered so fucking hard I am angy

File: code/game/objects/structures/crates_lockers/closets/coffin.dm [hunk 2]
```diff
@@ -51,4 +84,69 @@
...
+					)
+					drop_materials(drop_location())
+					qdel(src)
+			return
+
+		if(QUALITY_SAWING)
+			if(opened)
+				if(I.use_tool(user, src, WORKTIME_FAST, tool_type, FAILCHANCE_EASY, required_stat = STAT_MEC))
+					visible_message(
+							SPAN_NOTICE("\The [src] has been cut apart by [user] with \the [I]."),
```
Comment (Firefox13) [inline code/game/objects/structures/crates_lockers/closets/coffin.dm:138]: изображение
Add first person visible message. Usually third person messages start with the whoever does the thing

File: code/game/objects/structures/crates_lockers/closets/coffin.dm [hunk 3]
```diff
@@ -51,4 +84,69 @@
...
+							"You hear [tool_type]."
```
Comment (Firefox13) [inline code/game/objects/structures/crates_lockers/closets/coffin.dm:139]: "You hear sawing."
"You hear hammering."
"You hear prying."
Sound messages derived from tool qualities sound odd. How about:
"You hear wood sawing."
"You hear nails being pried out."
"You hear nails being hammered in."
Something like that.

Comment (Firefox13) [reply code/game/objects/structures/crates_lockers/closets/coffin.dm:139]: I *KNOW* it's already like that in some messages. Be better, put some effort.

File: code/game/objects/structures/crates_lockers/closets/coffin.dm [hunk 4]
```diff
@@ -51,4 +84,69 @@
...
+					)
+					drop_materials(drop_location())
+					qdel(src)
+				return
+		if(QUALITY_HAMMERING)
+			if(opened)
+				if(I.use_tool(user, src, WORKTIME_FAST, tool_type, FAILCHANCE_EASY, required_stat = STAT_MEC))
+					visible_message(
+							SPAN_NOTICE("\The [src] has had it's cover secured by [user] with \the [I]."),
```
Comment (Firefox13) [inline code/game/objects/structures/crates_lockers/closets/coffin.dm:148]:
```suggestion
							SPAN_NOTICE("\The [src] has had its cover secured by [user] with \the [I]."),
```

File: code/datums/craft/recipes/furniture.dm [hunk 5]
```diff
@@ -83,6 +83,13 @@
 		list(CRAFT_MATERIAL, 10, MATERIAL_WOOD),
 	)
 
+/datum/craft_recipe/furniture/bigcoffin
+	name = "pauper's coffin"
+	result = /obj/structure/closet/coffin/pauper
+	steps = list(
+		list(CRAFT_MATERIAL, 30, MATERIAL_WOOD),
```
Comment (Firefox13) [inline code/datums/craft/recipes/furniture.dm:90]:
```suggestion
		list(CRAFT_MATERIAL, 30, MATERIAL_WOOD)
```

File: code/game/objects/structures/crates_lockers/closets/coffin.dm (hunk 2 above)
Comment (Mycah142) [reply code/game/objects/structures/crates_lockers/closets/coffin.dm:138]: I followed other Closet based tool interactions, but I can change this, yes.

**Comment by Mycah142** [comment]
Alright. This is 99% the finished product. I tried to make the pyre flame gif better (I am barely a coder, nevermind an artist), I improved the messages, and I even made sure that anyone trapped in it can escape as easily as an unlocked closet so it isn't just easy round removal. Opening it causes the flames to go out as well.

The only thing missing is making a fire extinguisher work with it, but that is better tackled as a completely different PR where I add reagent interactions to structures (which currently doesn't exist, it's just snowflake code on shit like Thermite) so that the fire can be extinguished via, well, an extinguisher.

File: code/datums/craft/recipes/furniture.dm [hunk 6]
```diff
@@ -83,6 +83,13 @@
 		list(CRAFT_MATERIAL, 10, MATERIAL_WOOD),
 	)
 
+/datum/craft_recipe/furniture/bigcoffin
```
Comment (Firefox13) [inline code/datums/craft/recipes/furniture.dm:86]:
```suggestion
/datum/craft_recipe/furniture/pauper
```
The actual object's path is `pauper` too so let's keep it organized?

Comment (Mycah142) [reply code/datums/craft/recipes/furniture.dm:86]: Good point

File: code/game/objects/structures/crates_lockers/closets/coffin.dm [hunk 7]
```diff
@@ -36,6 +57,20 @@
 
 	return TRUE
 
+/obj/structure/closet/coffin/proc/pyre()
+	if(opened)
+		close() //If someone wanted an open casket funeral, we still need the casket to close to determine occupant
+	if(occupant && occupant.is_dead())
+		var/mob/N = key2mob(occupant.mind.key)
+		to_chat(N, SPAN_NOTICE("Your remains have been reduced to ash. Your crew respawn time has been reduced by [(COFFIN_RESPAWN_BONUS)/600] minutes."))
```
Comment (hyperioo) [inline code/game/objects/structures/crates_lockers/closets/coffin.dm:65]:
```suggestion
		to_chat(N, SPAN_NOTICE("Your remains have been reduced to ash. Your crew respawn time has been reduced by [round(COFFIN_RESPAWN_BONUS/(1 MINUTE))] minutes."))
```

Comment (hyperioo) [reply code/game/objects/structures/crates_lockers/closets/coffin.dm:65]: COFFIN_RESPAWN_BONUS			25 MINUTES
It's expressed in minutes.

**Pull request merged**